src/ocnicepost.fd/utils_mod.F90

```diff
@@ -772,6 +773,17 @@ subroutine write_grib2_2d(fname, gcf, dims, nflds, field, vfill)
             where ( field(:,n) .ne. vfill ) field(:,n) = field(:,n) + 273.15
          endif
 
+         if (trim(gcf(n)%name_gb2) == 'THFLX') then
+            idx = findloc(trim(gcf(:)%name_gb2) == 'NSWRF', .true.)
+            if (idx > 0) then
+               field(:,n) = field(:,n) + field(:,idx)
```

Comment:
Should this be wrapped in a `where` statement, similar to how `ICETMP` is assigned on line 773?

```suggestion
               where ( field(:, n) .ne. vfill .and. field(:, idx) .ne vfill ) field(:, n) = field(:, n) + field(:, idx)